CHANGELOG.md

```diff
@@ -4,6 +4,20 @@
 
 ### Other changes
 
+* **API**  
+    + The submit endpoint will now present all errors in an array:
```

Comment:
Can we run this change past a fireclouder? It seems like it might be more obnoxious for them than it would appear to us...

Reply:
What was it before?

Reply:
Well our generic "transform an exception in a json response" was already doing that if the exception was an `AggregatedException`, but the logic that looks at the form parameters in the request was not accumulated errors and throwing at the first one so it the `errors` array was never there.

Reply:
Got it. This is demonstrably better. As long as we make it clear that FC will need to update it's fine.

Reply:
To clarify further:

Up until now, the logic processing the submit request is wrapped in a `Try`, and the first thing that throws becomes the single exception we record. We then fail the request using this exception which results in something like 
```
{
  "status": "fail",
  "message": "Error(s): blablabla"
}
```

Note that because errors are not accumulated, there can't possibly be more than one in the message.

Now the same logic that transforms this exception into the above json (which we use for all endpoints), is smart enough to detect if the exception is an `AggregatedException` and will then create an `errors` array which will contain all the "sub" exceptions.

This change makes use of that, but @cjllanwarne is right that in doing so the "message" field doesn't contain the error anymore, but rather a context message that pretty much says "your submit request failed, see the errors array to know why".

A couple options would be 

1) if there's actually only one sub exception, put it in the "message" field and only use the "errors" array when there is more.
2) put a big string with all the exceptions concatenated in message and forget about the "errors" array (although it can definitely already appear in other endpoints that throws aggregated exceptions)

Once again, since it is true that FC might have to do some changes to adapt to this, I'm totally fine with saying "let's keep this for later". I saw it as a "nice to have" while doing this but it's totally unrelated to the yaml stuff.

Reply:
Ok, got it now. FWIW the `cromwell.yaml` (ie the now source of truth for our API, right? right?!?) doesn't have it as an array so this would indeed be a change.

If option #2 is indeed viable let's either go forward as-is now or fall back to that. I'll double check on the FC side to see what they want to do.